src/Functions/transform.cpp

```diff
@@ -418,11 +413,11 @@ namespace
             PaddedPODArray<Out> & out_pod,
             const PaddedPODArray<Out> & to_pod,
             const Def & def,
-            const size_t size,
-            const UInt32 out_scale,
-            const UInt32 def_scale) const
+            size_t input_rows_count,
+            UInt32 out_scale,
+            UInt32 def_scale) const
```

Comment:
Why are the `const`s removed? In general I like to have the `const` qualifier (e.g. good protection against accidentaly modifying the arguments that shouldn't be modified. I wouldn't ask you in this review to add const everywhere, but if it is already there, wondering why to remove?

Reply:
There are two schools about passing by-const-value:
- the const school which argues that the function arguments cannot be modified
- the non-const school which argues that
   1. modifying function arguments has no effect on the outside world (i.e. the caller) because arguments are passed by-value, so it is "harmless"
   2. modifying function arguments actually allows re-using existing variables

I don't have a strong preference, yet the code in ClickHouse goes in 99% of all cases with the non-const school. For this reason (and only this), I removed const.